### ISSUE TITLE
`IS DISTINCT FROM` not parsed in Snowflake dialect

### ISSUE THREAD

**Issue by yusufshalaby** [sqlfluff/sqlfluff#7585]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff 4.0.4 with dialect = snowflake fails to parse IS DISTINCT FROM, which is valid Snowflake SQL.

### Expected Behaviour

Linting to pass when a SQL file includes `IS DISTINCT FROM`

### Observed Behaviour

"Found unparsable section" errors

### How to reproduce

Lint the following SQL file
```sql
select
    case
        when 'foo' is distinct from 'bar' then 'baz'
    end as result
from my_table
```

### Dialect

Snowflake

### Version

4.0.4

### Configuration

```toml
[sqlfluff]
dialect = snowflake
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by yusufshalaby** [comment]
Never mind this was referenced already 
https://github.com/sqlfluff/sqlfluff/issues/3284
